### ISSUE TITLE
document how to build project

### ISSUE THREAD

**Issue by lfzCarlosC** [openwhisk/openwhisk#800]
When I try to build project using gradle build under openwhisk home directory, it throws error build failed:

executing 'docker build -t whisk/javaaction E:\J2EE project\openwhisk\core\javaAction' failed ....

I use gradle eclipse to build project, building was successful but I do not know how to run test case under tests directory.It seems that each components has a dockerFile and should I created all docker instance for all  components and then I can run the test script?

**Comment by daisy-ycguo** [comment]
@lfzCarlosC 
I use "ant build" to build all projects.
The test command should be "gradle test". You can try.

@csantanapr I think we need a developer guide.

**Comment by rabbah** [comment]
Indeed - we lost a top level build, deploy, test in the documentation refactoring. It's now buried here https://github.com/openwhisk/openwhisk/tree/master/tools/vagrant/custom#build-and-deploy
and here https://github.com/openwhisk/openwhisk/tree/master/tools/vagrant/custom#running-tests.

Since we are in the process of removing `ant` and will need to update these documents, we should include a top level build/deploy/test summary.

@csantanapr am I missing some other place where the instructions may be located.

**Comment by lfzCarlosC** [comment]
@daisy-ycguo  
Could you tell me from which directory do you use to execute this? openwhisk home dir? and you execute command "vagrant ssh -- ant build" or "ant build" on host machine?

**Comment by csantanapr** [comment]
Closing since how to build under Vagrant is documented in [tools/vagrant/README.md](https://github.com/openwhisk/openwhisk/blob/master/tools/vagrant/README.md)
TLDR;
```
#build like this
cd openwhisk
./gradlew distDocker
```

**Comment by rabbah** [comment]
How about adding the tldr to the top level readme (and include deploy "fresh" as well)?

**Comment by csantanapr** [comment]
@rabbah what do you mean? to doc `wskdev fresh` in main README?

I'm brainstorming on how to best organize README.md, CONTRIBUTING.md and maybe create a new DEVELOPER.md

Want to keep README short and focus to an user that wants to run openwhisk locally but not interested on contributing or modifying the code. the fastest and easier way to get OW running locally and try it out.

Was looking at angular2 repo as a pattern, what you think?
[README.md](https://github.com/angular/angular/blob/master/README.md)
[CONTRIBUTING.md](https://github.com/angular/angular/blob/master/CONTRIBUTING.md)
[DEVELOPER.md](https://github.com/angular/angular/blob/master/DEVELOPER.md)
[COMMITTER.md](https://github.com/angular/angular/blob/master/COMMITTER.md)
